events/TaskBase.h

```diff
@@ -0,0 +1,163 @@
+/* events
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TASK_BASE_H
+#define TASK_BASE_H
+
+#include "platform/Callback.h"
+#include "platform/mbed_assert.h"
+#include "LinkEntry.h"
+
+namespace rtos {
+class Semaphore;
+}
+
+namespace events {
+/** \addtogroup events */
+
+
+class TaskQueue;
+
+/** TaskBase
+ *
+ *  Representation of a caller allocated task
+ * @ingroup events
+ */
+class TaskBase : public LinkEntry {
```

Comment:
I think it is possible to reduce responsibilities exposed by the base class. 
The wait mechanism can be exposed in a subclass as the event queue doesn't have to know about it. 

Also, why is there no `execute` function exposed ?